www/include/eventLogs/xml/data.php

```diff
@@ -616,20 +623,33 @@
     }
 
     $limitReq = "";
-    $limitReq2 = "";
     if ($export !== "1") {
-        $limitReq = " LIMIT " . $num * $limit . ", " . $limit;
+        $offset = $num * $limit;
+        $queryValues['offset'] = [\PDO::PARAM_INT => $offset];
+        $limitReq = " LIMIT :offset, :limit";
```

Comment:
what happen if $limit = 'toto' ?
the bind will try to set a "LIMIT XX, false". you need to check $limit consistency too and that limit is > 0. Maybe add a default value to 10

Reply:
Line 178 & 179 - Is it not enough ?
```
$limit = isset($inputs["limit"]) ? (int) $inputs["limit"] : 30;
$queryValues['limit'] = [\PDO::PARAM_INT => $limit];
```

Reply:
indeed, I missed line 178... I thought you were using line 91 : https://github.com/centreon/centreon/pull/8764/files/1456afa271f43908dc8a5433fa15644cc06e615c#diff-4d945a7087649f5660dfcfb5e045d1eeR91.
But, if $limit = 'toto', then (int) $limit == 0 and your query will return nothing instead of the 30
Why send the query then ?
Why keep two checks when only one is enough  (line 91 + 178) ?

Reply:
done